streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -1653,6 +1678,34 @@ private Map<String, Object> getClientCustomProps() {
         return props;
     }
 
+    public static Set<String> verifyTopologyOptimizationConfigs(final String config) {
+        final List<String> configs = Arrays.asList(config.split("\\s*,\\s*"));
+        final Set<String> verifiedConfigs = new HashSet<>();
+        // Verify it doesn't contain none or all plus a list of optimizations
+        if (configs.contains(NO_OPTIMIZATION) || configs.contains(OPTIMIZE)) {
+            if (configs.size() > 1) {
+                throw new ConfigException("A topology can either not be optimized with " + NO_OPTIMIZATION + " "
+                                              + "or optimized. If you want to optimize the "
+                                              + "topology, you can choose between all "
+                                              + "optimizations with " + OPTIMIZE + " " + "or "
+                                              + "specific optimizations by specifying a comma "
+                                              + "separated list.");
```

Comment:
I took a closer look at the text of this error, and I'm afraid it might be confusing to users. The rendered text will say "A topology can either not be optimized with none or optimized.", which seems hard to parse to me, both because the sentence structure is confusing and because the config to turn off optimizations is mentioned ("none"), but the config to turn them on is not ("all").

```suggestion
                throw new ConfigException(
                    "\"" + config + "\" is not a valid optimization config. " +
                    "Valid optimization configs are: " +
                    "\"" + OPTIMIZE + "\" to enable all optimizations, " +
                    "\"" + NO_OPTIMIZATION + "\" to disable all optimizations, " +
                    "or a comma separated list of specific optimizations: " +
                    "(\"" + REUSE_KTABLE_SOURCE_TOPICS + "\", " +
                    MERGE_REPARTITION_TOPICS + "\"). "
                );
```

Realistically, I'd probably build the string for the second part of that message ("Valid ...") once and save it in a private constant for use in both the doc string as well as the validation error.